### ISSUE TITLE
414 URI too long response despite maxInitializeLineKB set very high

### ISSUE THREAD

**Issue by cian-k** [linkerd/linkerd#2081]
# Thanks for your help improving the project! #

## Getting Help ##

Github issues are for bug reports and feature requests. For questions about
Linkerd, how to use it, or debugging assistance, start by
[asking a question in the forums](https://discourse.linkerd.io/) or join us on
[Slack](https://slack.linkerd.io/).

Full details at [CONTRIBUTING.md](https://github.com/linkerd/linkerd/blob/master/CONTRIBUTING.md).

## Filing a Linkerd issue ##

Issue Type:

- [X ] Bug report
- [ ] Feature request

**What happened**:
414 Response received from Linkerd when trying to URI with ~6500 characters.
Relevant properties set:
"maxInitialLineKB": 2000000

**What you expected to happen**:
Request to be passed through to destination

**How to reproduce it (as minimally and precisely as possible)**:
Set "maxInitialLineKB": 2000000 and hit Linkerd with a URI of ~6500 characters
@dadjeibaah managed to reproduce, more details in Slack: 
https://linkerd.slack.com/archives/C0JV5E7BR/p1532040497000011

**Anything else we need to know?**:

**Environment**:
- linkerd/namerd version, config files:
Linkerd version 1.4.0
- Platform, version, and config files (Kubernetes, DC/OS, etc):
Running on Kubernetes using official Docker release
- Cloud provider or hardware configuration:
AWS

**Comment by dadjeibaah** [comment]
@cian-k updating this issue with what we figured out. So, it seems like Linkerd uses the `maxInitialSizeKB` value as intended. The one thing you will need to ensure is that the service you are trying to reach is configured to accept requests with large initial lines. 

When reproducing this issue, the test server I used was configured to accept initial line sizes of 4KB. After setting the `maxInitialSize` to 20MB, the server returned an `HTTP 200` to my test client through Linkerd.
